packages/protocol/contracts/layer1/preconf/impl/PreconfSlasher.sol

```diff
@@ -0,0 +1,319 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "src/layer1/based/ITaikoInbox.sol";
+import "src/layer1/preconf/iface/IPreconfSlasher.sol";
+import "src/shared/common/EssentialContract.sol";
+import "src/layer1/preconf/libs/LibPreconfConstants.sol";
+import "src/shared/libs/LibStrings.sol";
+import "src/shared/libs/LibTrieProof.sol";
+import "solady/src/utils/LibRLP.sol";
+
+/// @title PreconfSlasher
+/// @custom:security-contact security@taiko.xyz
+contract PreconfSlasher is IPreconfSlasher, EssentialContract {
+    address public immutable urc;
+    ITaikoInbox public immutable taikoInbox;
+    uint64 public immutable l2ChainId;
+
+    uint256[50] private __gap;
+
+    constructor(
+        address _resolver,
+        address _urc,
+        address _taikoInbox
+    )
+        EssentialContract(_resolver)
+    {
+        urc = _urc;
+        taikoInbox = ITaikoInbox(_taikoInbox);
+        l2ChainId = taikoInbox.v4GetConfig().chainId;
+    }
+
+    function init(address _owner) external initializer {
+        __Essential_init(_owner);
+    }
+
+    /// @inheritdoc ISlasher
+    function slash(
```

Comment:
may I suggest to move most of the code into a view function, for example, validateSlash(),  so that it can be used to verify the inputs are all valid for slashing.